### PR TITLE
Fix IRIs

### PR DESCRIPTION
**Pull request by OR13**
Fix links

related https://github.com/perma-id/w3id.org/pull/2017

^ with this being merged redirects now work for JSON Schemas...

https://w3id.org/traceability/schemas/IntentToSell.json

### CODE REVIEW

**Comment by mprorock** [comment]
@OR13 let me know when this PR is complete and all tests passing so that we can start assigning issues out on individual corrections based on this PR.  We will also updated docs and examples to clear up misunderstandings around the json schema -> json-ld process

File: docs/test-vectors/IntentToSell.json [hunk 1]
```diff
@@ -237,7 +237,7 @@
         "sku": "163443407321"
       },
       "declarationDate": "1-1-2021",
-      "sellByDate": "10-2-2021"
+      "sellByDate": "10-3-2021"
```
Comment (TallTed) [inline docs/test-vectors/IntentToSell.json:240]:
```suggestion
      "sellByDate": "10-3-2021"
```
I don't understand how changing dates falls under fixing IRIs ... but as long as you're fixing dates, please let's use ISO standard YYYY-MM-DD across the board!  (Is the above meant to say March 10, i.e., is it DD-MM-YYYY, or October 3, i.e., is it MM-DD-YYYY?)

Comment (mprorock) [reply docs/test-vectors/IntentToSell.json:240]: per README:

> Wherever possible, dates should be formatted as YYYY-MM-DD so as to be directly compatible with xsd:date.

Comment (OR13) [reply docs/test-vectors/IntentToSell.json:240]: This stuff is all autogenerated... please open an issue to track this, I am not going to fix anything but links in this PR.

**Comment by OR13** [comment]
Tests are passing locally... unsure why they are failing in CI.

**Comment by mprorock** [comment]
> Tests are passing locally... unsure why they are failing in CI.

looks like some max line length issues on the lint

File: docs/test-vectors/IntentToSell.json (hunk 1 above)
Comment (TallTed) [reply docs/test-vectors/IntentToSell.json:240]: ...and yet, multiple dates have had their middle-element increased by 1.

Fine, new issue coming.

File: packages/test-vectors/src/generators/AgInspectionReport.js [hunk 2]
```diff
@@ -5,80 +5,80 @@ const { getInspector } = require('./Inspector');
...
-    };
-    return example;
+  const example = {
+    '@context': ['https://w3id.org/traceability/v1'],
+    type: 'AgInspectionReport',
+    facility,
+    inspector,
+    shipment,
+    applicant,
+    inspectionDate: `${inspectDate.getMonth()}-${inspectDate.getDay()}-${inspectDate.getFullYear()}`,
```
Comment (mprorock) [inline packages/test-vectors/src/generators/AgInspectionReport.js:81]:
```suggestion
    inspectionDate: `${inspectDate.getFullYear()}-${inspectDate.getMonth()}-${inspectDate.getDay()}`,
```

File: packages/test-vectors/src/generators/AgPackage.js [hunk 3]
```diff
@@ -1,59 +1,59 @@
...
-    };
-    return example;
+  const example = {
+    '@context': ['https://w3id.org/traceability/v1'],
+    type: 'AgPackage',
+    packageName: faker.commerce.productName(),
+    grade: foodGrades[faker.random.number({ min: 0, max: 4 })],
+    responsibleParty,
+    voicePickCode: faker.random.number({ min: 1000, max: 9999 }).toString(),
+    date: `${thisDate.getMonth()}-${thisDate.getDay() + 1}-${thisDate.getFullYear()}`,
```
Comment (mprorock) [inline packages/test-vectors/src/generators/AgPackage.js:53]:
```suggestion
    date: `${thisDate.getFullYear()}-${thisDate.getMonth()}-${thisDate.getDay() + 1}`,
```

File: packages/test-vectors/src/generators/ppq203.js [hunk 4]
```diff
@@ -4,37 +4,36 @@ const { getPlace } = require('./Place');
...
+  const sigDate = new Date(faker.date.recent());
+
+  const example = {
+    '@context': ['https://w3id.org/traceability/v1'],
+    type: 'ppq203',
+    certificateNumber: faker.random.number({ min: 10000000, max: 999999999999 }).toString(),
+    foreignPortExport,
+    carrierId: faker.random.number({ min: 10000000, max: 999999999999 }).toString(),
+    portOfEntry,
+    signatureDate: `${sigDate.getMonth()}-${sigDate.getDay()}-${sigDate.getFullYear()}`,
```
Comment (mprorock) [inline packages/test-vectors/src/generators/ppq203.js:33]:
```suggestion
    signatureDate: `${sigDate.getFullYear()}-${sigDate.getMonth()}-${sigDate.getDay() + 1}`,
```

**Review by mprorock — CHANGES_REQUESTED** [review]
some date generation updates

File: packages/test-vectors/src/generators/IntentToSell.js [hunk 5]
```diff
@@ -3,29 +3,29 @@ const { getProduct } = require('./Product');
...
-        declarationDate: dDate.getMonth() + "-" + dDate.getDay() + "-" + dDate.getFullYear(),
-        sellByDate: fDate.getMonth() + "-" + fDate.getDay() + "-" + fDate.getFullYear()
-    };
+  const example = {
+    '@context': ['https://w3id.org/traceability/v1'],
+    type: 'IntentToSell',
+    seller,
+    purchaser,
+    product,
+    declarationDate: `${dDate.getMonth()}-${dDate.getDay()}-${dDate.getFullYear()}`,
```
Comment (mprorock) [inline packages/test-vectors/src/generators/IntentToSell.js:24]:
```suggestion
    declarationDate: `${dDate.getFullYear()}-${dDate.getMonth()}-${dDate.getDay()}`,
```

File: packages/test-vectors/src/generators/IntentToSell.js [hunk 6]
```diff
@@ -3,29 +3,29 @@ const { getProduct } = require('./Product');
...
+    sellByDate: `${fDate.getMonth()}-${fDate.getDay()}-${fDate.getFullYear()}`,
```
Comment (mprorock) [inline packages/test-vectors/src/generators/IntentToSell.js:25]:
```suggestion
    sellByDate: `${fDate.getFullYear()-${fDate.getMonth()}-${fDate.getDay()}}`,
```

File: packages/test-vectors/src/generators/LEIevidenceDocument.js [hunk 7]
```diff
@@ -1,162 +1,162 @@
 const faker = require('faker');
 
 const getLEIevidenceDocument = () => {
-  const fakePast = new Date(faker.date.past())
-  const fakeFuture = new Date(faker.date.future())
-  const pastDate = fakePast.getMonth() + "-" + fakePast.getDay() + "-" + fakePast.getFullYear();
-  const futureDate = fakeFuture.getMonth() + "-" + fakeFuture.getDay() + "-" + fakeFuture.getFullYear();
+  const fakePast = new Date(faker.date.past());
+  const fakeFuture = new Date(faker.date.future());
+  const pastDate = `${fakePast.getMonth()}-${fakePast.getDay()}-${fakePast.getFullYear()}`;
```
Comment (mprorock) [inline packages/test-vectors/src/generators/LEIevidenceDocument.js:6]:
```suggestion
  const pastDate = `${fakePast.getFullYear()}-${fakePast.getMonth()}-${fakePast.getDay()}`;
```

File: packages/test-vectors/src/generators/LEIevidenceDocument.js [hunk 8]
```diff
@@ -1,162 +1,162 @@
...
+  const futureDate = `${fakeFuture.getMonth()}-${fakeFuture.getDay()}-${fakeFuture.getFullYear()}`;
```
Comment (mprorock) [inline packages/test-vectors/src/generators/LEIevidenceDocument.js:7]:
```suggestion
  const futureDate = `${fakeFuture.getFullYear()}-${fakeFuture.getMonth()}-${fakeFuture.getDay()}`;
```

File: packages/test-vectors/src/generators/ProductRegistrationEvidenceDocument.js [hunk 9]
```diff
@@ -4,38 +4,38 @@ const getProductRegistrationEvidenceDocument = () => {
...
-    "url": `https://example.com/${product}`,
-    "isBasedOn": faker.commerce.productName(),
-    "image": faker.image.imageUrl()
+    category: getCategory(),
+    inProductGroupWithID: `Group${faker.random.number({ min: 100, max: 999 })}`,
+    productID: `${faker.random.alphaNumeric(10).toUpperCase()}`,
+    mpn: `${faker.random.alphaNumeric(10).toUpperCase()}`,
+    gtin: `${faker.random.number({ min: 100000000000, max: 99999999999999 })}`, // 13-14 numeric characters
+    isAccessoryOrSparePartFor: 'n/a',
+    releaseDate: `${pastDate.getMonth()}-${pastDate.getDay()}-${pastDate.getFullYear()}`,
```
Comment (mprorock) [inline packages/test-vectors/src/generators/ProductRegistrationEvidenceDocument.js:23]:
```suggestion
    releaseDate: `${pastDate.getFullYear()}-${pastDate.getMonth()}-${pastDate.getDay()}`,
```

**Review by mprorock — CHANGES_REQUESTED** [review]
and some more dates

**Comment by OR13** [comment]
@mprorock @TallTed  please not more changes suggested for things unrelated to broken JSON-LD... I am happy fix other issues in a subsequent PR.

**Review by mprorock — APPROVED** [review]
and some more dates

**Comment by mprorock** [comment]
This is a critical issue to fix broken JSON-LD that also blocks other key PRs for folks, as well as their ability to reference working examples.  i am merging since it is passing tests

**Pull request merged**